internal/pkg/agent/cmd/apply_flavor.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/install"
+	"github.com/elastic/elastic-agent/internal/pkg/cli"
+	v1 "github.com/elastic/elastic-agent/pkg/api/v1"
+)
+
+func newApplyFlavorCommandWithArgs(_ []string, streams *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "apply-flavor",
+		Short: "Apply Flavor cleans up unnecessary components from agent installation directory",
+		Run: func(c *cobra.Command, _ []string) {
+			if err := applyCmd(); err != nil {
+				fmt.Fprintf(streams.Err, "Error: %v\n%s\n", err, troubleshootMessage())
+				logExternal(fmt.Sprintf("%s apply flavor failed: %s", paths.BinaryName, err))
+				os.Exit(1)
+			}
+		},
+		Hidden: true,
+	}
+
+	return cmd
+}
+
+func applyCmd() error {
+	topPath := paths.Top()
+	detectedFlavor, err := install.UsedFlavor(topPath, "")
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+	if detectedFlavor == "" {
```

Comment:
Same here, we should log that we detected no flavor and did nothing.

In general this function should log what it did, neither install.Flavor or install.ApplyFlavor seem to log anything either. We can't tell "successfully changed flavor" from "did nothing because the flavor was already applied" for example.